### PR TITLE
Drop support for legacy Python 2.7

### LINKED ISSUE #650: Drop python 2.7 support
We should consider dropping support for python 2.7, as it will be unsupported soon...

### PR DESCRIPTION
**Pull request by hugovk**
Fixes #650.

Adds `python_requires` to help pip.

Also upgrade syntax with https://github.com/asottile/pyupgrade and unit tests to use more useful asserts.

### CODE REVIEW

File: androguard/core/bytecodes/axml/__init__.py [hunk 1]
```diff
@@ -1084,12 +1084,12 @@ def _fix_value(self, value):
         if not self.__charrange or not self.__replacement:
             if sys.maxunicode == 0xFFFF:
```
Comment (reox) [inline androguard/core/bytecodes/axml/__init__.py]: Do you know if there are narrow 3.x builds? otherwise we can remove this too...

**Comment by reox** [comment]
Wow, this looks very nice!
To summarize what you did:
* replace all __super__(Classname, self.)... with __super__()
* remove all (object) from class definitions
* replace C string fmt with propper .format
* add regex type string if not present
* remove all unicode string definitions
* change set() to {}

There are also import from __future__ and builtins which could be removed now, as they were primary for the py2/3 compatability.
Also the one comment: This was a fix for narrow 2.x builds, but I think there are no narrow py3 builds? At least according to SO (https://stackoverflow.com/questions/14790708/do-unicode-strings-in-python-3-still-depend-on-narrow-wide-builds) this only affects up to 3.2. But as our minimal version is 3.4 (due to other issues) we should be fine.

This PR sounds like, that the next version will be 4.0.0 ;)

**Comment by reox** [comment]
@androguard/external-contributors please have a look!

File: androguard/core/bytecodes/axml/__init__.py (hunk 1 above)
Comment (hugovk) [reply androguard/core/bytecodes/axml/__init__.py]: I don't know.

**Comment by eighthave** [comment]
Most of the changes look simple and good.  Changing the formats could cause breakage, so I hope those cases are covered by tests.

**Comment by reox** [comment]
> so I hope those cases are covered by tests.

so far, the tests work for mutf8 parsing - which was one of the most annoying things to have both working for py2 and py3.

**Pull request merged**